src/Mono.Android/Java.Interop/JavaConvert.cs

```diff
@@ -314,44 +314,74 @@ public static T FromJavaObject<T>(IJavaObject? value, out bool set)
 				using (var v = new Java.Lang.Boolean ((bool) value))
 					return JNIEnv.ToLocalJniHandle (v);
 			} },
+			{ typeof (bool[]), value => {
+				return JNIEnv.NewArray ((bool[]) value);
+			} },
 			{ typeof (byte),   value => {
 				using (var v = new Java.Lang.Byte ((sbyte) (byte) value))
 					return JNIEnv.ToLocalJniHandle (v);
 			} },
+			{ typeof (byte[]), value => {
```

Comment:
My concern with this approach is that I don't think it'll be "recursive" enough.  This handles `List<byte[]>`, but what happens when we hit a `List<byte[][]>`?  Or a `List<byte[][][]>`? Or a …?

Reply:
…which is why in [my original comment](https://github.com/xamarin/xamarin-android/issues/4098#issuecomment-572818288) I said:

> but we might want to instead consider using following the "recursive type" approach used around `JNIEnv.NewArray(Array, Type)`.